### PR TITLE
Added immutable to never modified variables in several files

### PR DESCRIPTION
**Pull request by JackStouffer**
The way I see it, it's best practice to add either `const` or `immutable` everywhere it can be added. It acts not only as a sanity check, but as documentation to the future maintainer.

### CODE REVIEW

File: src/object.d [hunk 1]
```diff
@@ -275,7 +275,7 @@ class TypeInfo
     /// Swaps two instances of the type.
     void swap(void* p1, void* p2) const
     {
-        size_t n = tsize;
+        immutable n = tsize;
```
Comment (wilzbach) [inline src/object.d]: You kept the type for all the other changes.

**Comment by wilzbach** [comment]
> feel free to merge.

I think @JackStouffer isn't part of the Druntime team, so he can't merge PRs here (same for me).

**Comment by JackStouffer** [comment]
Ping

**Comment by JackStouffer** [comment]
Ping

**Comment by CyberShadow** [comment]
Oops, this should have been merged a long time ago :/

**Comment by wilzbach** [comment]
I will save this & rebase it tomorrow.

**Comment by wilzbach** [comment]
> I will save this & rebase it tomorrow.

Actually it wasn't much effort -> done now ;-)

**Pull request merged**